policy/handler.go

```diff
@@ -218,3 +291,46 @@ func (h *Handler) updateHandler(current, next *Policy) {
 		h.ticker = time.NewTicker(next.EvaluationInterval)
 	}
 }
+
+// enforceCooldown blocks until the cooldown period has been reached, or the
```

Comment:
we should probably document it somewhere, but one of the really nice things about the way that we block here and the fact that we use `NewTicker` for `h.ticker` is the following: `h.ticker` will "drop items on the floor" as the ticks come in. our approach doesn't stop or alter the ticker, so it continues to tick while we are blocked in the cooldown. this means that if `EvaluationInterval < Cooldown`, we will immediately trigger an evaluation after exiting cooldown (as in the previous PR). however, if `Cooldown < EvaluationInterval`, then cooldown doesn't have any effect; but that seems semantically sound.

Reply:
also, because `true/false` are ambiguous returns, i've seen (mostly in other languages) where folks recommend "named" values (sometimes even protected by type safety, apparently not in golang) for readability. like so: 
https://play.golang.org/p/WRfAPGDAJCF
is this something that folks do in golang? is there a more idiomatic way to do this?

Reply:
I am not the biggest fan of bool returns like this. I had considered using an error to identify if the cooldown completed without interruption, but this is just checked and wouldn't be logged so seems incorrect.

The function could use a named return to aid readers:
```
func (h *Handler) enforceCooldown(ctx context.Context, t time.Duration) (completed bool) {
...

	select {
	case <-timer.C:
                completed = true
		return
	case <-ctx.Done():
		return
	case <-h.doneCh:
		return
	}
}
```

Reply:
that's definitely better. mainly just a lazy gripe on my part that one has to go look at the function docs/code to see whether caller should look for `true` or `false`